### PR TITLE
Introduced two version support feature

### PR DESCRIPTION
**Pull request by hlts2**
Now that Vald can support multiple versions, clients should also be able to support multiple versions.
In this PR, I implemented a workflow to support multiple versions.
The main logic is implemented as a common workflow in [vald-client-ci](https://github.com/vdaas/vald-client-ci), so I implemented a call to it.

## Summary by CodeRabbit

- **New Features**
  - Introduced automated workflows for end-to-end testing, release handling, and periodic synchronization of the project.
  
- **Enhancements**
  - Updated the `.gitignore` to better manage project files and dependencies.
  - Improved `Makefile` with new targets and updated paths for more efficient builds and testing.
  
- **Bug Fixes**
  - Adjusted plugin versions in `buf.gen.yaml` to ensure compatibility and stability.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,93 +1,32 @@
...
-    if: ${{ success() || failure()  }}
-    steps:
-      - uses: actions/checkout@v3
-      - uses: ./.github/actions/notify-slack
-        with:
-          author_name: Release vald-java-client
-          slack_notify_webhook_url: ${{ secrets.SLACK_WEBHOOK_URL }}
+    # TODO: change to the following before main merge.
+    # uses: vdaas/vald-client-ci/.github/workflows/_release.yaml@main
+    uses: vdaas/vald-client-ci/.github/workflows/_release.yaml@feat/add-common-workflows
```
Comment (hlts2) [inline .github/workflows/release.yml:29]:
```suggestion
    # uses: vdaas/vald-client-ci/.github/workflows/release.yaml@main
    uses: vdaas/vald-client-ci/.github/workflows/release.yaml@feat/add-common-workflows
```

File: Makefile [hunk 2]
```diff
@@ -111,73 +132,74 @@ vald/sha/print:
...
+	curl -L https://raw.githubusercontent.com/rinx/word2vecjson/master/data/wordvecs1000.json -o $(TEST_DATASET_PATH)
 
-.PHONY: vald/java/version/stable
-## print JAVA_LTS_STABLE_VERSTION value
-vald/java/version/stable:
-	@cat $(JAVA_LTS_STABLE_VERSTION)
+.PHONY: ci/deps/install
+## install deps for CI environment
+ci/deps/install:
+	@echo "Nothing do be done"
```
Comment (hlts2) [inline Makefile:162]:
```suggestion
	@echo "Nothing to do"
```

File: Makefile [hunk 3]
```diff
@@ -111,73 +132,74 @@ vald/sha/print:
...
-	    else \
-	        new_version="$${vald_version}"; \
-	    fi; \
-	    echo "VALD_VERSION: $${vald_version}, NEW_CLIENT_VERSION: $${new_version}"; \
-	    echo "$${new_version}" > version/VALD_CLIENT_JAVA_VERSION)
-	sed -i -e "s/^version = ".*"\$$/version = \"`cat version/VALD_CLIENT_JAVA_VERSION`\"/" build.gradle
+.PHONY: ci/deps/update
+## update deps for CI environment
+ci/deps/update:
+	@echo "Nothing do be done"
```
Comment (hlts2) [inline Makefile:167]:
```suggestion
	@echo "Nothing to do"
```

**Comment by hlts2** [comment]
@coderabbitai review

File: script/prepare_gradle_properties.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (hlts2) [inline script/prepare_gradle_properties.sh:1]: > [!NOTE]
> This logic was implemented in the makefile, but if the environment variable contains special characters, execution will fail, so it was changed to be implemented in a this script file.

**Review by kmrmt — APPROVED** [review]

**Pull request merged**